### PR TITLE
Update tplink docs for auth and tapo

### PR DESCRIPTION
**Pull request by sdb9696**
## Proposed change

Updated documentation relating to [core PR](https://github.com/home-assistant/core/pull/105143) enabling newer Kasa and Tapo devices.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/105143
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by bdraco** [comment]
I think we can remove 
```

## Unsupported devices
### Plugs
- KP125M (supported via [Matter](/integrations/matter/#tp-link-tapo-p125m-power-plug), but without energy monitoring features)
- 
```

and add it to supported since it works now and I'm using it

File: source/_integrations/tplink.markdown [hunk 1]
```diff
@@ -24,9 +24,12 @@ ha_quality_scale: platinum
...
+You need to provision your newly purchased device to connect to your network before it can be added via the integration. This can be done either by using [kasa command-line tool](https://python-kasa.readthedocs.io/en/latest/cli.html#provisioning) or by adding it to the official Kasa or Tapo app before trying to add them to Home Assistant.
+
+If your device is a newer Kasa or Tapo device it will require your TP-Link cloud username and password to authenticate for local access.
+If you have an older device that does not currently require authentication and you want it to stay that way, then you should disable automatic firmware updates, otherwise it may upgrade to firmware that requires it.
 
 There is currently support for the following device types within Home Assistant:
```
Comment (rytilahti) [inline source/_integrations/tplink.markdown]: I'm not sure if listing the supported platforms (just below this sentence) is anymore relevant, so it should be removed altogether to clean it up?

File: source/_integrations/tplink.markdown [hunk 2]
```diff
@@ -96,13 +101,39 @@ See [Supported Devices in python-kasa](https://github.com/python-kasa/python-kas
...
+- L900-10
+
 
 ## Unsupported devices
 
-### Plugs
+Other devices may work but if you encuonter issues submit a bug report to [python-kasa](https://github.com/python-kasa/python-kasa).
+
+Other bulbs may also work, but with limited color temperature range (2700-5000).
+If you find a bulb isn't reaching the full-color temperature boundaries, submit a bug report to [python-kasa](https://github.com/python-kasa/python-kasa).
```
Comment (rytilahti) [inline source/_integrations/tplink.markdown]: I think we could also remove this, as I think the coverage is pretty good, the tapo bulbs deliver this information via the API, and we still log a warning on unknown models.

File: source/_integrations/tplink.markdown [hunk 3]
```diff
@@ -96,13 +101,39 @@ See [Supported Devices in python-kasa](https://github.com/python-kasa/python-kas
...
+### Light strips
+
+- L900-5
+- L900-10
+
 
 ## Unsupported devices
 
-### Plugs
+Other devices may work but if you encuonter issues submit a bug report to [python-kasa](https://github.com/python-kasa/python-kasa).
```
Comment (rytilahti) [inline source/_integrations/tplink.markdown]:
```suggestion
Other devices may work but if you encounter issues submit a bug report to [python-kasa](https://github.com/python-kasa/python-kasa).
```

I'd also move this on top of the listings and reword it accordingly. Then we can remove the whole "unsupported devices" subsection altogether.

File: source/_integrations/tplink.markdown [hunk 4]
```diff
@@ -55,14 +58,16 @@ See [Supported Devices in python-kasa](https://github.com/python-kasa/python-kas
 - KP125
 - KP401
 - EP10
-- EP25 (HW versions 2.5 and earlier)
+- EP25 (hw_version < 2.6)
```
Comment (rytilahti) [inline source/_integrations/tplink.markdown]:
```suggestion
- EP25 (Hardware version < 2.6)
```

File: source/_integrations/tplink.markdown [hunk 5]
```diff
@@ -96,13 +101,39 @@ See [Supported Devices in python-kasa](https://github.com/python-kasa/python-kas
 - KL420
 - KL430
 
-Other bulbs may also work, but with limited color temperature range (2700-5000). If you find a bulb isn't reaching the full-color temperature boundaries, submit a bug report to [python-kasa](https://github.com/python-kasa/python-kasa).
+## Supported devices requiring authentication
+
+### Plugs
+
+- EP25 (hw_version >= 2.6)
```
Comment (rytilahti) [inline source/_integrations/tplink.markdown]:
```suggestion
- EP25 (Hardware version >= 2.6)
```

File: source/_integrations/tplink.markdown [hunk 6]
```diff
@@ -96,13 +101,39 @@ See [Supported Devices in python-kasa](https://github.com/python-kasa/python-kas
...
+- KP125M
+- P110
+- HS100 (UK hw_version = 4.1 fw_version = 1.1.0)
```
Comment (rytilahti) [inline source/_integrations/tplink.markdown]:
```suggestion
- HS100 (UK, hardware version 4.1 with firmware 1.1.0)
```

File: source/_integrations/tplink.markdown [hunk 7]
```diff
@@ -40,7 +43,7 @@ There is currently support for the following device types within Home Assistant:
 
 See [Supported Devices in python-kasa](https://github.com/python-kasa/python-kasa#supported-devices) for an up to date list.
```
Comment (rytilahti) [inline source/_integrations/tplink.markdown:40]: Move the note about unsupported versions here.

File: source/_integrations/tplink.markdown [hunk 8]
```diff
@@ -40,7 +43,7 @@ There is currently support for the following device types within Home Assistant:
...
 
-## Supported devices
+## Supported devices not requiring authentication
```
Comment (rytilahti) [inline source/_integrations/tplink.markdown]: Maybe we should indent these to become below the "supported devices" section above? This way we could simplify these to become "No authentication required" and "Requires authentication" or something similar.

File: source/_integrations/tplink.markdown [hunk 9]
```diff
@@ -24,9 +24,12 @@ ha_quality_scale: platinum
...
 
-You need to provision your newly purchased device to connect to your network before it can be added via the integration. This can be done either by using [kasa command-line tool](https://python-kasa.readthedocs.io/en/latest/cli.html#provisioning) or by adding it to the official Kasa app before trying to add them to Home Assistant. If you use the app, do not upgrade the firmware if it presents the option to avoid blocking the local access by potential firmware updates.
+You need to provision your newly purchased device to connect to your network before it can be added via the integration. This can be done either by using [kasa command-line tool](https://python-kasa.readthedocs.io/en/latest/cli.html#provisioning) or by adding it to the official Kasa or Tapo app before trying to add them to Home Assistant.
+
+If your device is a newer Kasa or Tapo device it will require your TP-Link cloud username and password to authenticate for local access.
+If you have an older device that does not currently require authentication and you want it to stay that way, then you should disable automatic firmware updates, otherwise it may upgrade to firmware that requires it.
```
Comment (rytilahti) [inline source/_integrations/tplink.markdown]:
```suggestion
If you have an older device that does not currently require authentication, you may consider disabling automatic firmware updates to keep it that way.
```
Just a suggestion to make it simpler.

File: source/_integrations/tplink.markdown [hunk 10]
```diff
@@ -96,13 +101,39 @@ See [Supported Devices in python-kasa](https://github.com/python-kasa/python-kas
...
 ## Unsupported devices
 
-### Plugs
+Other devices may work but if you encuonter issues submit a bug report to [python-kasa](https://github.com/python-kasa/python-kasa).
+
+Other bulbs may also work, but with limited color temperature range (2700-5000).
+If you find a bulb isn't reaching the full-color temperature boundaries, submit a bug report to [python-kasa](https://github.com/python-kasa/python-kasa).
 
-- KP125M (supported via [Matter](/integrations/matter/#tp-link-tapo-p125m-power-plug), but without energy monitoring features)
+## Light strip effects
```
Comment (rytilahti) [inline source/_integrations/tplink.markdown:127]: Maybe we should pre-emptively document that these are only supported on non-authenticated devices for the time being?

File: source/_integrations/tplink.markdown (hunk 1 above)
Comment (sdb9696) [reply source/_integrations/tplink.markdown]: Done

File: source/_integrations/tplink.markdown (hunk 2 above)
Comment (sdb9696) [reply source/_integrations/tplink.markdown]: Done

File: source/_integrations/tplink.markdown (hunk 3 above)
Comment (sdb9696) [reply source/_integrations/tplink.markdown]: Done

File: source/_integrations/tplink.markdown (hunk 4 above)
Comment (sdb9696) [reply source/_integrations/tplink.markdown]: Done

File: source/_integrations/tplink.markdown (hunk 5 above)
Comment (sdb9696) [reply source/_integrations/tplink.markdown]: Done

File: source/_integrations/tplink.markdown (hunk 6 above)
Comment (sdb9696) [reply source/_integrations/tplink.markdown]: Done

File: source/_integrations/tplink.markdown (hunk 7 above)
Comment (sdb9696) [reply source/_integrations/tplink.markdown:40]: Done

File: source/_integrations/tplink.markdown (hunk 8 above)
Comment (sdb9696) [reply source/_integrations/tplink.markdown]: Done

File: source/_integrations/tplink.markdown (hunk 9 above)
Comment (sdb9696) [reply source/_integrations/tplink.markdown]: Done

File: source/_integrations/tplink.markdown (hunk 10 above)
Comment (sdb9696) [reply source/_integrations/tplink.markdown:127]: Done

**Comment by sdb9696** [comment]
All comments addressed

**Review by rytilahti — DISMISSED** [review]
LGTM, thanks! :1st_place_medal:

File: source/_integrations/tplink.markdown [hunk 11]
```diff
@@ -90,19 +92,40 @@ See [Supported Devices in python-kasa](https://github.com/python-kasa/python-kas
...
+
+#### Bulbs
+
+- L510B
+- L530E
+
+#### Light strips
+
+- L900-5
+- L900-10
```
Comment (bdraco) [inline source/_integrations/tplink.markdown:123]:
```suggestion
- L900-10
- L920
```

Screenshot 2024-01-22 at 11 34 55 AM


confirmed working..  I think I added a fixture for this one as well already

Comment (bdraco) [reply source/_integrations/tplink.markdown:123]: https://github.com/python-kasa/python-kasa/blob/master/kasa/tests/fixtures/smart/L920-5(US)_1.0_1.1.0.json

**Review by bdraco — DISMISSED** [review]

**Review by bdraco — APPROVED** [review]

**Pull request merged**